### PR TITLE
chore(pipelines): support retrieving all pipelines in workspace

### PR DESCRIPTION
**Pull request by dannyrandall**
Changes `ListPipelines` to get legacy pipelines as well as new pipelines stored in `copilot/pipelines/*/manifest.yml`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by efekarakus — APPROVED** [review]
😍 great test cases!

File: internal/pkg/workspace/workspace.go [hunk 1]
```diff
@@ -173,68 +178,64 @@ type PipelineManifest struct {
...
+	for _, dir := range files {
+		if dir.IsDir() {
+			addManifest(filepath.Join(pipelinesDir, dir.Name(), manifestFileName))
 		}
 	}
-	return pipelineManifests, nil
+
+	// sort manifests alphabetically by Name
+	sort.Slice(manifests, func(i, j int) bool {
+		return manifests[i].Name < manifests[j].Name
```
Comment (huanjani) [inline internal/pkg/workspace/workspace.go:235]: Slick!

File: internal/pkg/workspace/workspace_test.go [hunk 2]
```diff
@@ -520,63 +520,135 @@ func TestWorkspace_ListPipelines(t *testing.T) {
...
+				fs.Mkdir("/copilot", 0755)
+				fs.Mkdir("/copilot/pipelines", 0755)
+				fs.Mkdir("/copilot/pipelines/beta", 0755)
+				fs.Mkdir("/copilot/pipelines/prod", 0755)
+
+				return fs
+			},
+			wantedPipelines: nil,
+			wantedErr:       nil,
+		},
```
Comment (huanjani) [inline internal/pkg/workspace/workspace_test.go:652]: Totally my bad, but can we add some unit tests for catching errs?

File: internal/pkg/workspace/workspace.go [hunk 3]
```diff
@@ -7,23 +7,28 @@
 //  .
 //  ├── copilot                        (application directory)
 //  │   ├── .workspace                 (workspace summary)
-//  │   └── my-service
+//  │   ├── my-service
```
Comment (huanjani) [inline internal/pkg/workspace/workspace.go:10]: Thanks for changing this! Checkbox checked! 😉

File: internal/pkg/workspace/workspace.go [hunk 4]
```diff
@@ -173,68 +178,64 @@ type PipelineManifest struct {
...
-	// Look for legacy pipeline.
-	legacyPath, err := ws.PipelineManifestLegacyPath()
-	if err != nil {
-		return nil, err
-	}
-	manifest, err := ws.ReadPipelineManifest(legacyPath)
-	if err != nil {
+	var manifests []PipelineManifest
+
+	addManifest := func(manifestPath string) {
```
Comment (huanjani) [inline internal/pkg/workspace/workspace.go:183]: Thanks for extracting this functionality.

**Review by huanjani — COMMENTED** [review]
This looks fantastic!
Just a one thing while we wait to see if the design changes tomorrow. 😸

File: internal/pkg/workspace/workspace_test.go (hunk 2 above)
Comment (dannyrandall) [reply internal/pkg/workspace/workspace_test.go:652]: ok I added some tests to cover as many errors as possible, not sure if they are over the top or not :) let me know if you want to keep all of the new tests or not.

Comment (huanjani) [reply internal/pkg/workspace/workspace_test.go:652]: Love the tests. Thank you!

**Review by huanjani — APPROVED** [review]
Looks great!

**Pull request merged**